Analysis/src/SignalObjectSelector.cpp

```diff
@@ -8,40 +8,42 @@ namespace analysis {
 
 SignalObjectSelector::SignalObjectSelector(SignalMode _mode) : mode(_mode)
 {
-    if(mode == SignalMode::HTT || mode == SignalMode::HTT_sync){
-	DR2_leptons = std::pow(cuts::H_tautau_2016::DeltaR_betweenSignalObjects, 2);
-	discriminator = TauIdDiscriminator::byIsolationMVArun2017v2DBoldDMwLT2017;
-    }
-    else if(mode == SignalMode::TauPOG_default){
-    	DR2_leptons = std::pow(cuts::H_tautau_2016::DeltaR_betweenSignalObjects, 2);
-	discriminator = TauIdDiscriminator::byIsolationMVArun2017v2DBoldDMwLT2017;
-    }
-    else if(mode == SignalMode::TauPOG_deepTauVsJet || mode == SignalMode::TauPOG_deepTauVsJet_full){
-	DR2_leptons = std::pow(cuts::H_tautau_2016::DeltaR_betweenSignalObjects, 2);
-	discriminator = TauIdDiscriminator::byDeepTau2017v2p1VSjet;
-    }
-    else if(mode == SignalMode::HH_legacy){
-    	DR2_leptons = std::pow(cuts::hh_bbtautau_2017::DeltaR_betweenSignalObjects, 2);
-	discriminator = TauIdDiscriminator::byIsolationMVArun2017v2DBoldDMwLT2017;
-    }
-    else if(mode == SignalMode::HH){
-    	 DR2_leptons = std::pow(cuts::hh_bbtautau_2017::DeltaR_betweenSignalObjects, 2);
-	     discriminator = TauIdDiscriminator::byDeepTau2017v2p1VSjet;
-    }
+    if(mode == SignalMode::HTT || mode == SignalMode::HTT_sync)
```

Comment:
and `DR2_leptons` for the other modes? also if mode is not found the exception should be thrown.